server/etcdserver/txn/txn.go

```diff
@@ -242,31 +242,35 @@ func Txn(ctx context.Context, lg *zap.Logger, rt *pb.TxnRequest, txnModeWriteWit
 	isWrite := !IsTxnReadonly(rt)
 	// When the transaction contains write operations, we use ReadTx instead of
 	// ConcurrentReadTx to avoid extra overhead of copying buffer.
-	var txnWrite mvcc.TxnWrite
+	var mode mvcc.ReadTxMode
 	if isWrite && txnModeWriteWithSharedBuffer /*a.s.Cfg.ExperimentalTxnModeWriteWithSharedBuffer*/ {
-		txnWrite = mvcc.NewReadOnlyTxnWrite(kv.Read(mvcc.SharedBufReadTxMode, trace))
+		mode = mvcc.SharedBufReadTxMode
 	} else {
-		txnWrite = mvcc.NewReadOnlyTxnWrite(kv.Read(mvcc.ConcurrentReadTxMode, trace))
+		mode = mvcc.ConcurrentReadTxMode
 	}
```

Comment:
nitpick on coding style.. 

```golang
mode := mvcc.ConcurrentReadTxMode
if isWrite && txnModeWriteWithSharedBuffer /*a.s.Cfg.ExperimentalTxnModeWriteWithSharedBuffer*/ {
    mode = mvcc.SharedBufReadTxMode
}
```

Reply:
What style? Go style suggest avoiding `else` statements in separate main vs alternative code paths. However, this doesn't apply to normal conditions.